nemo_skills/pipeline/utils.py

```diff
@@ -266,6 +277,100 @@ def get_packager():
         )
 
 
+def get_env_variables(cluster_config):
+    """
+    Will get the environment variables from the cluster config and the user environment.
+
+    The following items in the cluster config are supported:
+    - `required_env_vars` - list of required environment variables
+    - `env_vars` - list of optional environment variables
+
+    Args:
+        cluster_config: cluster config dictionary
+
+    Returns:
+        dict: dictionary of environment
+    """
+    env_vars = {}
+    # Check for user requested env variables
+    required_env_vars = cluster_config.get("required_env_vars", [])
+    for env_var in required_env_vars:
+        if env_var not in os.environ:
+            raise ValueError(f"Required environment variable {env_var} not found.")
+
+        env_vars[env_var] = os.environ[env_var]
+        logging.info(f"Adding required environment variable {env_var} (value={os.environ[env_var]})")
+
+    # Add optional env variables
+    optional_env_vars = cluster_config.get("env_vars", [])
+    for env_var in optional_env_vars:
+        if env_var in os.environ:
+            logging.info(f"Adding optional environment variable {env_var} (value={os.environ[env_var]})")
+            env_vars[env_var] = os.environ[env_var]
+        else:
+            logging.info(f"Optional environment variable {env_var} not found in user environment; skipping.")
+
+    return env_vars
+
+
+def get_mounts_from_config(cluster_config: dict, env_vars: dict = None):
+    """
+    Determines if there are mount paths that are being passed via environment variables.
+    Selects the key in the cluster config called `mounts` which is a list of strings.
+    Each string is in the format of `<str | {env_var}>:<str | {env_var}>` where `env_var`
+    is the name of the environment variable.
+
+    Args:
+        cluster_config (dict): cluster config dictionary
+        env_vars (dict): dictionary of environment variables
+
+    Returns:
+        list: updated list of mounts
+    """
+    mounts = cluster_config.get('mounts', [])
+
+    # if env_vars is None, we will get the env_vars from the cluster config
+    if env_vars is None:
+        env_vars = get_env_variables(cluster_config)
+
+    # if there are env_mounts, we will add the mounts from the env_mounts
+    for mount_id in range(len(mounts)):
+        mount = mounts[mount_id]
+
+        if ":" not in mount:
+            raise ValueError(f"Invalid mount format: {mount}. The mount path must be separated by a colon.")
+
+        mount_source, mount_target = mount.split(":")
+
+        if mount_source[0] == "{" and mount_source[-1] == "}":
+            # Resolve the environment variable for the mount source
+            mount_source = mount_source[1:-1]
+
+            if mount_source not in env_vars:
```

Comment:
We don't need to restrict it to what in the config here, right? Let's just check os.environ for that, and the env_vars field will be there only to pass env vars inside of a job

Reply:
Hmm technically we dont need to restrict. But its very easy to miss the value. So ill instead check for the value from os.environ and raise an error if it doesn't exist there.